### PR TITLE
Set EventSource version for all events

### PR DESCRIPTION
**Pull request by jeffkl**
Fixes #

### Context
In https://github.com/dotnet/msbuild/pull/7390 I modified some of the event source methods but I just noticed that things look wrong in the latest internal preview.  It seems like Start/Stop event methods need to have the same version in order to be linked up.

image
image

I'm working on event messages like this in NuGet and just used a constant to rev the version of all event methods at the same time which seems like a better practice in general.

## Changes Made
Add a constant to represent the event method version and use that version in all event methods.  

### Testing
Manually tested locally, these are the events logged now:

image

Here is a sample for others to verify as well:

[PerfViewData.etl.zip](https://github.com/dotnet/msbuild/files/8364311/PerfViewData.etl.zip)

### CODE REVIEW

**Comment by jeffkl** [comment]
I think I'll never touch the event source in MSBuild ever again, somehow I keep messing it up...

File: src/Framework/MSBuildEventSource.cs [hunk 1]
```diff
@@ -8,11 +8,16 @@ namespace Microsoft.Build.Eventing
...
-    /// Changes to existing event method signatures will not be reflected unless you update the <see cref="EventAttribute.Version" /> property or assign a new event ID.
+    /// Changes to existing event method signatures will not be reflected unless you update the <see cref="MSBuildEventSource.EventVersion" /> constant or assign a new event ID.
     /// </remarks>
     [EventSource(Name = "Microsoft-Build")]
     internal sealed class MSBuildEventSource : EventSource
     {
+        /// <summary>
+        /// Represents the version of the events.  Be sure to increment this any time you modify an existing event method.
+        /// </summary>
+        public const int EventVersion = 100;
```
Comment (Forgind) [inline src/Framework/MSBuildEventSource.cs:19]: Is there some reason you starting at 100 instead of 1? Does the version comparer think 9 > 10?

**Review by Forgind — COMMENTED** [review]
Does this first version update update any method signature changes that already exist? If so, I'd say it makes sense to just update the version of a particular event whenever we need to. This way leads to extra events being updated unnecessarily.

File: src/Framework/MSBuildEventSource.cs (hunk 1 above)
Comment (Forgind) [reply src/Framework/MSBuildEventSource.cs:19]: Ping 😄

**Comment by jeffkl** [comment]
I spent a few hours trying to understand ETW more and when and when not to use this Version attribute metadata and couldn't find anything obvious.  I don't have any more time to spend on this so I'm going to close this for now since my solution doesn't seem desirable.

**Pull request closed without merging**